### PR TITLE
[14.0] edi_webservice: add generic send component

### PR DESCRIPTION
**Pull request by simahawk**
Depends on:

- [x] OCA/web-api#7
- [x] OCA/edi#647

TODO:

- [x] update docs

### CODE REVIEW

File: edi_webservice_oca/tests/__init__.py [hunk 1]
```diff
@@ -1 +1 @@
-from . import test_edi_webservice
+from . import test_backend
```
Comment (simahawk) [inline edi_webservice_oca/tests/__init__.py:1]: missing import here

File: edi_webservice_oca/tests/common.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    def _setup_records(cls):
+        super()._setup_records()
+        cls.filedata = base64.b64encode(b"This is a simple file")
+        vals = {
+            "model": cls.partner._name,
+            "res_id": cls.partner.id,
+            "exchange_file": cls.filedata,
+        }
+        cls.record = cls.backend.create_record("test_csv_output", vals)
+        cls.sender = cls.backend._find_component(
```
Comment (simahawk) [inline edi_webservice_oca/tests/common.py]: useless var

File: edi_webservice_oca/tests/test_send.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+          send:
+            usage: webservice.send
+            work_ctx:
+              webservice:
+                method: post
+                kwargs:
+                  url_params:
+                    endpoint: push/here
+        """
+        cls.record.type_id.set_settings(cls.settings1)
```
Comment (simahawk) [inline edi_webservice_oca/tests/test_send.py:42]: this cannot work w/o #647

**Review by SilvioC2C — APPROVED** [review]

**Comment by SilvioC2C** [comment]
Overall LGTM, @simahawk might wanna take a look to the failing tests

**Comment by simahawk** [comment]
> Overall LGTM, @simahawk might wanna take a look to the failing tests

it has been fixed in #647

**Pull request merged**